activation/post_verifier.go

```diff
@@ -102,7 +160,9 @@ func (v *postVerifier) Verify(
 type postVerifierOpts struct {
 	opts           PostProofVerifyingOpts
 	prioritizedIds []types.NodeID
-	autoscaling    bool
+	autoscaling    *struct {
+		postStates postStatesGetter
+	}
```

Comment:
NIT: why the embedded struct? `postStates` can also be checked for `nil`.

```suggestion
	postStates postStatesGetter
```

Reply:
Previously, there was a bool to enable/disable autoscaling. Now it also needs the interface to work. I think the code is more readable with:
```go
if autoscaling != nil {
    // enable autoscaling
}
```
compared to 
```go
if postStates != nil {
    // enable autoscaling
}
```

Plus, comparing interface against nil doesn't always work.

Reply:
The comparison against `nil` for interfaces is only relevant if you can asing a typed `nil` to that field:

```go
var autoscaling myInterface = nil

autoscaling == nil // true

var value *myType = nil
autoscaling = value

autoscaling == nil // false (because it was assigned a nil `myType`)
```

I personally think anonymous embedded structs don't necessarily make the code easier readable; `postStates` is a prerequisite for autoscaling; so if it is not set there is no autoscaling